aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -518,7 +518,7 @@ WriteFpropResultsVectorized(
     if (offset >= shift && offset < size) {
       output[offset] = epilogue(input[offset]);
     }
-    size -= blockDim.x;
+    size -= blockDim.x > size ? size : blockDim.x;
```

Comment:
Here the size decrement should be same as input/output increment.

Reply:
Same as above.